src/sklearn_evaluation/plot/roc.py

```diff
@@ -154,57 +126,51 @@ class ROCAdd(Plot):
-        self.figure = plt.figure()
-        ax = self.figure.add_subplot()
-        added_curve_label = "ROC curve 2"
+        self.a = a
+        self.b = b
 
-        if hasattr(a, "roc_rates_n_classes"):
-            self.ax = _plot_roc_multi_classification(
-                a.fpr, a.tpr, a.roc_rates_n_classes, ax
-            )
-        else:
-            _plot_roc(a.fpr, a.tpr, ax=ax)
+    def plot(self, ax=None):
+        a = self.a
+        b = self.b
 
-        if hasattr(b, "roc_rates_n_classes"):
-            self.ax = _plot_roc_multi_classification(
-                b.fpr, b.tpr, b.roc_rates_n_classes, ax, curve_label=added_curve_label
-            )
+        if ax is None:
+            _, ax = plt.subplots()
+
+        _generate_plot_from_fpr_tpr_lists(a.fpr, a.tpr, ax, labels=a.labels)
+
+        # mark the added curve with 2
+        if b.labels is not None and len(b.labels) > 0:
+            b_labels = [label + " 2" for label in b.labels]
         else:
-            _plot_roc(b.fpr, b.tpr, ax=ax, curve_label=added_curve_label)
+            b_labels = ["ROC curve 2"]
+
+        _generate_plot_from_fpr_tpr_lists(b.fpr, b.tpr, ax, labels=b_labels)
 
+        self.ax_ = ax
+        self.figure_ = ax.figure
 
-class ROC(Plot):
+        return self
+
+
+class ROC(AbstractPlot):
     """
     Plot ROC curve
     Parameters
     ----------
-    y_true : array-like, shape = [n_samples]
-        Correct target values (ground truth).
-
-    y_score : array-like, shape = [n_samples] or [n_samples, 2] for binary
-        classification or [n_samples, n_classes] for multiclass
-        Target scores (estimator predictions).
-
-    fpr : ndarray of shape (>2,), default: None
+    fpr : ndarray of shape (>2,), list of lists or list of numbers
         Increasing false positive rates such that element i is the false
-        positive rate of predictions with score >= `thresholds[i]`. If
-        None, it will be calculated based on y_true and y_score.
+        positive rate of predictions with score >= `thresholds[i]`.
 
-    tpr : ndarray of shape (>2,), default: None
+    tpr : ndarray of shape (>2,), list of lists or list of numbers
         Increasing true positive rates such that element `i` is the true
-        positive rate of predictions with score >= `thresholds[i]`. If
-        None, it will be calculated based on y_true and y_score.
+        positive rate of predictions with score >= `thresholds[i]`.
+
+    labels : list of str, default: None
+        Set curve labels
 
     ax: matplotlib Axes, default: None
         Axes object to draw the plot onto, otherwise uses current Axes
 
-    Notes
-    -----
-    It is assumed that the y_score parameter columns are in order.
-    For example, if ``y_true = [2, 2, 1, 0, 0, 1, 2]``, then the
-    first column in y_score must contain the scores for class 0,
-    second column for class 1 and so on.
-
     Examples
     --------
     .. plot:: ../examples/roc_new_api.py
```

Comment:
let's update this example so it shows how the new API works (using from_raw_data).

- roc for binary classification
- roc for multi-class
- comparing two binary classifiers (plot1 + plot2)

Reply:
We already had these examples (except the multi-class).
I changed the names and added an example for multi-class.

please check if these examples are ok.

@idomic Does this multi-class example address #98 ?

Reply:
Not really, the issue there is the error message.